core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -565,12 +569,40 @@ class PartKeyLuceneIndex(ref: DatasetRef,
                            partId: Int, startTime: Long, endTime: Long): Document = {
     val partKeyBytesRef = new BytesRef(partKeyOnHeapBytes, partKeyBytesRefOffset, partKeyNumBytes)
     luceneDocument.get().reset(partId, partKeyBytesRef, startTime, endTime)
+
+    // If configured and enabled, Multi-column facets will be created on "partition-schema" columns
+    createMultiColumnFacets(partKeyOnHeapBytes, partKeyBytesRefOffset)
+
     cforRange { 0 until numPartColumns } { i =>
       indexers(i).fromPartKey(partKeyOnHeapBytes, bytesRefToUnsafeOffset(partKeyBytesRefOffset), partId)
     }
     luceneDocument.get().document
   }
 
+  // Multi-column facets to be created on "partition-schema" columns
+  private def createMultiColumnFacets(partKeyOnHeapBytes: Array[Byte], partKeyBytesRefOffset: Int): Unit = {
+    schema.options.multiColumnFacets.foreach(facetCols => {
+      facetCols match {
+        case (name, cols) => {
+          val concatFacetValue = cols.flatMap { col =>
+            val (columnInfo, pos) = schema.columnIdxLookup(col)
```

Comment:
If the column is missing it will throw ``java.util.NoSuchElementException``. Better option would be invoke ``get`` to get an ``Option``  and when ``None``, either throw a specific exception with meaningful message or log a warning and return empty string. Throwing an exception will fail adding the document to index.

Reply:
👍 fixed it